plugins/modules/lxd_storage_pool_info.py

```diff
@@ -0,0 +1,342 @@
+
+    def _get_storage_pool_info(self, pool_name):
+        """Get detailed information about a specific storage pool."""
+        url = self._build_url(f"/1.0/storage-pools/{pool_name}")
+        resp_json = self.client.do("GET", url, ok_error_codes=[404])
+
+        if resp_json["type"] == "error":
+            if resp_json.get("error_code") == 404:
+                self.module.fail_json(msg=f'Storage pool "{pool_name}" not found')
+            else:
+                self.module.fail_json(
+                    msg=f'Failed to retrieve storage pool "{pool_name}": {resp_json.get("error", "Unknown error")}'
+                )
+
+        return resp_json.get("metadata", {})
+
+    def get_storage_pools(self):
+        """Retrieve storage pool information based on module parameters."""
+        storage_pools = []
+
+        if self.name:
+            # Get information about a specific storage pool
+            pool_info = self._get_storage_pool_info(self.name)
+            storage_pools.append(pool_info)
+        else:
+            # Get information about all storage pools
+            pool_names = self._get_storage_pool_list()
+            for pool_name in pool_names:
+                pool_info = self._get_storage_pool_info(pool_name)
+                storage_pools.append(pool_info)
+
+        # Filter by type if specified
+        if self.pool_type:
+            storage_pools = [
+                pool for pool in storage_pools if pool.get("driver") in self.pool_type
+            ]
+
+        return storage_pools
+
+    def run(self):
+        """Run the main method."""
+        try:
+            if self.trust_password is not None:
+                self.client.authenticate(self.trust_password)
+
+            storage_pools = self.get_storage_pools()
+
+            result_json = {"storage_pools": storage_pools}
+            if self.client.debug:
+                result_json["logs"] = self.client.logs
+            self.module.exit_json(**result_json)
+        except LXDClientException as e:
+            fail_params = {"msg": e.msg, "changed": False}
+            if self.client.debug:
+                fail_params["logs"] = e.kwargs.get("logs", [])
+            self.module.fail_json(**fail_params)
+
+
+def main():
+    """Ansible Main module."""
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type="str"),
+            type=dict(type="list", elements="str"),
+            project=dict(type="str"),
+            url=dict(type="str", default=ANSIBLE_LXD_DEFAULT_URL),
+            snap_url=dict(
+                type="str", default="unix:/var/snap/lxd/common/lxd/unix.socket"
+            ),
+            client_key=dict(type="path", aliases=["key_file"]),
+            client_cert=dict(type="path", aliases=["cert_file"]),
+            trust_password=dict(type="str", no_log=True),
+        ),
+        supports_check_mode=True,
+    )
+
+    lxd_info = LXDStoragePoolInfo(module=module)
+    lxd_info.run()
```

Comment:
Just an idea (not necessarily a good one). Maybe we could wrap this block with:
```suggestion
    try:
        lxd_info = LXDStoragePoolInfo(module=module)
        lxd_info.run()
    except LXDClientException as e:
        fail_params = {"msg": e.msg}
        if module._verbosity >= 4:
            fail_params["logs"] = e.kwargs.get("logs", [])
        self.module.fail_json(**fail_params)
```
and then remove the `try`/`except` blocks in lines 228-236 and 302-316.